tket2-hseries/src/extension/result.rs

```diff
@@ -161,11 +161,27 @@ impl ResultOpDef {
     }
 
     fn result_signature(&self) -> SignatureFunc {
-        PolyFuncType::new(
-            [vec![TypeParam::String], self.type_params()].concat(),
-            Signature::new(self.arg_type(), type_row![]),
+        if self.is_array_result_op() {
+            // Arrays need to be returned to preserve no-implicit copy guarantees.
+            PolyFuncType::new(
+                [vec![TypeParam::String], self.type_params()].concat(),
+                Signature::new(self.arg_type(), vec![self.arg_type()]),
+            )
```

Comment:
Remove the vec![] for consistency (handled by Into).

Consider also just defining `let output = if ... else ...;` and then moving the shared code for the returned value after the conditional. Keeps the branched code shorter and removes duplication.

```suggestion
                Signature::new(self.arg_type(), self.arg_type()),
```